ksqldb-engine/src/main/java/io/confluent/ksql/engine/SandboxedExecutionContext.java

```diff
@@ -160,7 +160,11 @@ public ExecuteResult execute(
     ).execute(ksqlPlan.getPlan());
 
     // Having a streams running in a sandboxed environment is not necessary
-    result.getQuery().map(QueryMetadata::getKafkaStreams).ifPresent(streams -> streams.close());
+    if (!getKsqlConfig().getBoolean(KsqlConfig.KSQL_SHARED_RUNTIME_ENABLED)) {
```

Comment:
We need make sure all the runtimes for this sandbox gets closed not just the one for the new query. They will not be used again